### DISCUSSION TITLE
Request for Mermaid inline diagram support

### DISCUSSION THREAD

**Discussion by frehberg** [rust-lang/crates.io#5724] [Feature Requests]
AFAICS, github markdown supports mermaid inline diagrams, for example in README.md or this posting (do you see the  rendered mermaid graph below?)

[https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/ ](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/)

Does crate info page at crates.io support mermaid inline diagrams as well, if generated from project's  README.md ?

Regarding _docs as code_, rendering  the mermaid graphs of project's README.md in crates.io  would be really nice! 

    ```mermaid
         graph TD;
              mermaid -- supported-by --> rust-docs(rust docs aquamarine plugin);
              mermaid -- supported-by --> github;
              mermaid -- supported-by --> gitlab;
              mermaid -- supported-by --> reveal.JS;
              mermaid -- supported-by --> vue.JS;
              mermaid -- ? --> crates.io(crates.io markdown rendering pipeline); 
    ```

```mermaid
     graph TD;
          mermaid -- supported-by --> rust-docs(rust docs aquamarine plugin);
          mermaid -- supported-by --> github;
          mermaid -- supported-by --> gitlab;
          mermaid -- supported-by --> reveal.JS;
          mermaid -- supported-by --> vue.JS;
          mermaid -- ? --> crates.io(crates.io markdown rendering pipeline); 
```

**Comment by Turbo87** [comment]
This is currently not supported, sorry.

**Reply by Odonno** [reply]
Thank you for this @Turbo87, it works really great.

**Reply by Turbo87** [reply]
all thanks goes to @ToBinio, who figured out a way to implement this. also, there is still https://github.com/rust-lang/crates.io/issues/6786, which needs to be solved before it is fully supported :)

**Comment by frehberg** [comment]
It would be great if crates.io would support mermaid inline diagrams when scanning the specified markdown file (README.md by default)
Please join this request.

**Reply by fhaefemeier** [reply]
I would like to extend this request: Please support Kroki instead of a single diagram format. [Kroki](https://kroki.io) is the umbrella service to support a lot of different diagram formats.

**Comment by Odonno** [comment]
Yes. I would love that too. However, if this is a complex feature to implement, it could be possible to hide it from the crates.io readme web page? Instead of having something ugly or nothing, I prefer nothing. You could maybe parse the code section with a custom property and remove the entire code section, like this:

```
-```mermaid,hide
Example of a mermaid diagram
-```
```

This way, you can check if `hide` is present in the list for this code section and hide it.

(I added the extra "-" so you can see the code, this is not part of my idea).
